test/unit/internal/oidc/test_ambient.py

```diff
@@ -337,17 +325,18 @@ def test_gcp_impersonation_identity_token_request_timeout(monkeypatch):
     monkeypatch.setattr(ambient, "logger", logger)
 
     access_token = pretend.stub()
-    get_resp = pretend.stub(
-        raise_for_status=lambda: None, json=lambda: {"access_token": access_token}
-    )
-    post_resp = pretend.stub(raise_for_status=pretend.raiser(Timeout))
-    requests = pretend.stub(
-        get=pretend.call_recorder(lambda url, **kw: get_resp),
-        post=pretend.call_recorder(lambda url, **kw: post_resp),
-        HTTPError=HTTPError,
-        Timeout=Timeout,
-    )
-    monkeypatch.setattr(ambient, "requests", requests)
+    get_resp = pretend.stub(status=200, json=lambda: {"access_token": access_token})
+
+    def _request(meth, *a, **kw):
+        if meth == "GET":
+            return get_resp
+        elif meth == "POST":
+            raise ValueError
+        else:
+            assert False
+
+    u3 = pretend.stub(request=_request, exceptions=pretend.stub(MaxRetryError=ValueError))
```

Comment:
could you avoid exceptions arg here by raising MaxRetryError in the first place?

Reply:
> could you avoid exceptions arg here by raising MaxRetryError in the first place?

Unfortunately not: this happens because I stub out the entire `urllib3` package, so `urllib3.exeptions.MaxRetryError` would no longer be the "same" type. I could probably do a more precise stub instead, but `MaxRetryError` is also an annoying one to raise in tests (since it needs to be initialized with a URL, connection pool, response, etc).